### PR TITLE
osd: clean up process_peering_events

### PR DESCRIPTION
**Pull request by 363921219**
Signed-off-by: Jie Wang <jie.wang@kylin-cloud.com>

### CODE REVIEW

**Comment by dachary** [comment]
jenkins test this please (eio/scrub failure)

**Comment by liewegas** [comment]
1. Are you sure the assert is valid?
2. Can we drop the curmap assignement inside the loop instead?  Or, move curmap scope into the loop?

**Comment by dachary** [comment]
jenkins test this please (flake8 related error, now fixed)

**Comment by 363921219** [comment]
> Are you sure the assert is valid?
sure, "process_peering_events" will not be called if pgs is empty.
> Can we drop the curmap assignement inside the loop instead?
osdmap may be updated when 'process_peering_events' run,  so i am not sure it is valid.
 >Or, move curmap scope into the loop?
curmap is need outside the loop

**Comment by tchaikov** [comment]
i concur with @363921219, `ThreadPool` won't pass an empty list to `BatchWorkQueue` if `_void_dequeue()` returns NULL.

and `dispatch_context(rctx, 0, curmap, &handle)` out side of the loop uses the last `curmap` passed to the last call `advance_pg()`. so we can compare the `curmap.epoch` with the `nextmap.epoch` before sending the messages out.

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -9030,7 +9030,8 @@ void OSD::process_peering_events(
 {
   bool need_up_thru = false;
   epoch_t same_interval_since = 0;
-  OSDMapRef curmap = service.get_osdmap();
+  assert(!pgs.empty());
```
Comment (tchaikov) [inline src/osd/OSD.cc]: we can move this assert up into PeeringWQ::_process()`.

Comment (tchaikov) [inline src/osd/OSD.cc]: and also put more details into the commit message: `clean up` is too vague a word, IMHO.

**Comment by 363921219** [comment]
@tchaikov thanks for your comments

**Comment by tchaikov** [comment]
could you remove "Reviewed-by: Kefu Chai <kchai@redhat.com>" from your commit messages? it will be added to the merge commit by the one who merges your changes.

**Comment by tchaikov** [comment]
and please squash these two commit into a single one.

**Comment by dachary** [comment]
test this please (asok error now fixed in master)

**Comment by dachary** [comment]
jenkins test this please (eio)

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2016-11-22_09:44:32-rados-wip-kefu-testing---basic-mira/

**Pull request merged**